### DISCUSSION TITLE
Is it possible to insert images to CTkTextbox?

### DISCUSSION THREAD

**Discussion by Gasiro** [TomSchimansky/CustomTkinter#2496] [Q&A] [answered]
Trying to figure it out but don't know how to do it. 
Got this: "AttributeError: embedding images is forbidden, because would be incompatible with scaling"

**Comment by dipeshSam** [comment, marked as answer]
Access internal text box of CTkTextbox. Reason of so is mentioned in the [answer of one of the earlier discussion](https://github.com/TomSchimansky/CustomTkinter/discussions/2463#discussioncomment-9733801).

Sample code:
``` Python 

from tkinter import *
from customtkinter import CTkTextbox
from PIL.Image import open as open_
from PIL.ImageTk import PhotoImage



if __name__ == "__main__":
	app = Tk()
	
	tb = CTkTextbox(app)
	tb.pack(padx=50, pady=50)
	
	tb.insert("1.0", "Some Text")
	
	tk_img = PhotoImage(open_("theImage.png"))
	
	tb._textbox.image_create("1.1", image=tk_img)


	app.mainloop()
```

It will be helpful.
Regards.

**Reply by Gasiro** [reply]
it is so weird that I did inserted something here but the image is not showing.
<img width="534" alt="Screenshot 2024-07-05 at 14 46 52" src="https://github.com/TomSchimansky/CustomTkinter/assets/117976338/0ca6e3eb-0995-43ed-8c45-245ddc7e339c">
<img width="843" alt="Screenshot 2024-07-05 at 14 49 22" src="https://github.com/TomSchimansky/CustomTkinter/assets/117976338/dfb5587d-1b25-4bd0-853b-805c9be2a368">

**Reply by dipeshSam** [reply]
**Try to set your `insert_img` global.**

``` Python
global insert_img

```

**Reply by Gasiro** [reply]
it works very well. Thank you 
<img width="556" alt="Screenshot 2024-07-05 at 23 41 02" src="https://github.com/TomSchimansky/CustomTkinter/assets/117976338/892215ae-0a1c-435c-afcd-605c8566d3f3">

**Reply by dipeshSam** [reply]
The reason of so is: to display an image in Tkinter, the image must remain in memory. The Python garbage collector may delete the image to free up memory. Therefore, it is necessary to explicitly instruct Python to retain the image. Typically, local variables are collected as garbage once their scope is exited. _One way_ to prevent this is by using global variables to store the image.

You are welcome. Continue your great Tkinter journey😊

Regards.
